### PR TITLE
Cleaning up console output from tests

### PR DESCRIPTION
**Pull request by kraenhansen**
## What, How & Why?
I was getting back to Realm JS and got annoyed that the output from the tests are bloated with console.logs and output from the Realm JS client itself. It turned out the `Realm.Sync` loggers got reset before each test executed (as `Realm.clearTestState` was called): To ensure the logger is restored, this PR monkey-patches `Realm.clearTestState` to reset the logger and log level to appropriate levels.

## ☑️ ToDos
* [x] 📝 Changelog entry
* [x] 📝 `Compatibility` label is updated or copied from previous entry
* [x] 🚦 Tests
* [x] 📝 Public documentation PR created or is not necessary
* [x] 💥 `Breaking` label has been applied or is not necessary

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -146,6 +146,11 @@ Iterative development requires performing more of the steps manually:
 
 After making changes to the C++ source files rerun `npm run build-changes` to rebuild the files which have changed. Changes to JS or TS files don't require any manual steps beyond rerunning the tests.
 
+If you want the Realm JS Sync client's logs in the console, run your command with the `DEBUG=realm` environment variable sat (i.e. `DEBUG=realm npm run js-tests`)
+To adjust the lowest log level set the `REALM_LOG_LEVEL` environment variable (defaults to `info`).
+
+Some tests have been instrumented with detailed logging, run your command with the `DEBUG=tests:*` environment variable sat to see these (i.e. `DEBUG=tests:* npm run js-tests`)
```
Comment (bmunkholm) [inline CONTRIBUTING.md]:
```suggestion
Some tests have been instrumented with detailed logging, run your command with the `DEBUG=tests:*` environment variable set to see these (i.e. `DEBUG=tests:* npm run js-tests`)
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -146,6 +146,11 @@ Iterative development requires performing more of the steps manually:
 
 After making changes to the C++ source files rerun `npm run build-changes` to rebuild the files which have changed. Changes to JS or TS files don't require any manual steps beyond rerunning the tests.
 
+If you want the Realm JS Sync client's logs in the console, run your command with the `DEBUG=realm` environment variable sat (i.e. `DEBUG=realm npm run js-tests`)
```
Comment (bmunkholm) [inline CONTRIBUTING.md]:
```suggestion
If you want the Realm JS Sync client's logs in the console, run your command with the `DEBUG=realm` environment variable set (i.e. `DEBUG=realm npm run js-tests`)
```

File: tests/js/session-tests.js [hunk 3]
```diff
@@ -22,7 +22,11 @@
 
 /* global REALM_MODULE_PATH */
 
+// Run these tests with the `DEBUG=tests:session` enrivonment variable set to get the stdout of sub-processes.
```
Comment (bmunkholm) [inline tests/js/session-tests.js]:
```suggestion
// Run these tests with the `DEBUG=tests:session` environment variable set to get the stdout of sub-processes.
```

**Comment by kraenhansen** [comment]
The failing tests looks like a brittle test harness to me:
- The Node on Windows 8 seem to have a hard time finding the NodeJs executable
- The two React Native example apps seem to have a hard time connecting to the metro / development server.

**Comment by blagoev** [comment]
@kraenhansen  yeah. lets merge it

**Pull request merged**